lib/ex_integrate/step.ex

```diff
@@ -0,0 +1,23 @@
+defmodule ExIntegrate.Step do
+  @moduledoc """
+  Represents a step in the CI pipeline. 
+  """
+
+  @enforce_keys [:name, :command]
+  defstruct [:name, :command, args: [], command_data: nil]
+
+  @type t :: %__MODULE__{
+          args: [String.t()],
+          command: :String.t(),
+          command_data: Rambo.t(),
```

Comment:
@garrettmichaelgeorge what will `command_data` be used for? Just curious not saying I am opposed to it.